### PR TITLE
[IMP] payment_razorpay: fix processing data in token flow

### PR DESCRIPTION
**Pull request by kcv-odoo**
### [FIX] payment_razorpay: fix proccessing data in token flow
Steps:
- Install razorpay and sales app.
- Enable Razorpay provider.
- Create a create a token in razorpay.
- Pay via that token.

Issue:
- When we are paying via token we really don't need
processing values because those processing values
only needed when we open Razorpay form so here it is
calling APIs(customer and order) and creating order
unwantedly.

Fix:
- Return empty dict when operation is `online_token` in
`_get_specific_processing_values` method

IMP/REV of PR: https://github.com/odoo/odoo/pull/157533

task-3653372

### CODE REVIEW

**Comment by kcv-odoo** [comment]
Hello @AntoineVDV,

This is PR to backport [PR](https://github.com/odoo/odoo/pull/148673)
Also reverting/improving [Fix](https://github.com/odoo/odoo/pull/157533) and fix multiple order creation for [online token flow](https://discord.com/channels/968074811777642587/968074893889527898/1220717087547457657)
Could you check?

@hamo-odoo @jva-odoo could you check first commit changes and test everything working fine in POS after this changes

Thank you :smiley:

File: addons/payment_razorpay/models/payment_transaction.py [hunk 1]
```diff
@@ -34,8 +34,10 @@ def _get_specific_processing_values(self, processing_values):
         if self.provider_code != 'razorpay':
             return res
 
-        customer_is_required = self.tokenize or self.operation in ('online_token', 'offline')
-        customer_id = customer_is_required and self._razorpay_create_customer()['id']
+        if self.operation == 'online_token':
```
Comment (AntoineVDV) [inline addons/payment_razorpay/models/payment_transaction.py:37]:
```suggestion
        if self.operation in ('online_token', 'offline'):
```

File: addons/payment_razorpay/models/payment_transaction.py [hunk 2]
```diff
@@ -53,7 +55,7 @@ def _razorpay_create_customer(self):
         payload = {
             'name': self.partner_name,
             'email': self.partner_email,
-            'contact': self._validate_phone_number(self.partner_phone),
+            'contact': self.partner_phone and self._validate_phone_number(self.partner_phone) or "",
```
Comment (AntoineVDV) [inline addons/payment_razorpay/models/payment_transaction.py:58]:
```suggestion
            'contact': self.partner_phone and self._validate_phone_number(self.partner_phone) or '',
```

**Review by AntoineVDV — APPROVED** [review]
Seems to work fine! Just a small detail to fix :point_down: 

@robodoo delegate+

**Comment by kcv-odoo** [comment]
Thank you @AntoineVDV for review :sweat_smile: 

@robodoo r+

**Comment by kcv-odoo** [comment]
Just rebase the branch because runbot was red :slightly_frowning_face: 

@robodoo r+

**Pull request closed without merging**